substrate/bin/node-template/pallets/template/src/lib.rs

```diff
@@ -1,105 +1,188 @@
+	/// A storage item for this pallet.
+	///
+	/// In this template, we are declaring a storage item called `Something` that stores a single
+	/// `u32` value. Learn more about runtime storage here: <https://docs.substrate.io/build/runtime-storage/>
 	#[pallet::storage]
-	#[pallet::getter(fn something)]
-	// Learn more about declaring storage items:
-	// https://docs.substrate.io/main-docs/build/runtime-storage/#declaring-storage-items
 	pub type Something<T> = StorageValue<_, u32>;
 
-	// Pallets use events to inform users when important changes are made.
-	// https://docs.substrate.io/main-docs/build/events-errors/
+	/// Events that functions in this pallet can emit.
+	///
+	/// Events are a simple means of indicating to the outside world (such as dApps, chain explorers
+	/// or other users) that some notable update in the runtime has occurred. In a FRAME pallet, the
+	/// documentation for each event field and its parameters is added to a node's metadata so it
+	/// can be used by external interfaces or tools.
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
-		/// Event documentation should end with an array that provides descriptive names for event
-		/// parameters. [something, who]
-		SomethingStored { something: u32, who: T::AccountId },
+		/// A user has successfully set a new value.
+		SomethingStored {
+			/// The new value set.
+			something: u32,
+			/// The account who set the new value.
+			who: T::AccountId,
+		},
 	}
 
-	// Errors inform users that something went wrong.
+	/// Errors that can be returned by this pallet.
+	///
+	/// Errors tell users that something went wrong so it's important that they are informative.
+	/// Similar to events, error documentation is added to a node's metadata so it's important that
+	/// they have helpful documentation associated with them.
+	///
+	/// Note: the `Error` enum my always be generic over `<T>`, which is expanded to `<T: Config>`.
 	#[pallet::error]
 	pub enum Error<T> {
-		/// Error names should be descriptive.
+		/// The value retrieved was `None` as no value was previously set.
 		NoneValue,
-		/// Errors should have helpful documentation associated with them.
+		/// There was an attempt to increment the value in storage over `u32::MAX` (4294967295).
 		StorageOverflow,
 	}
 
-	// Dispatchable functions allows users to interact with the pallet and invoke state changes.
-	// These functions materialize as "extrinsics", which are often compared to transactions.
-	// Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+	/// The pallet's dispatchable functions.
+	///
+	/// Dispatchable functions allows users to interact with the pallet and invoke state changes.
+	/// These functions materialize as "extrinsics", which are often compared to transactions.
+	/// They must always return a `DispatchResult` and be annotated with a weight and call index.
```

Comment:
If you want to add more links for more detailed information of the macros.

More info in the call index annotation here: https://paritytech.github.io/substrate/master/frame_support_procedural/attr.call_index.html and the weight here: https://paritytech.github.io/substrate/master/frame_support_procedural/attr.weight.html